ratelimits/limiter.go

```diff
@@ -294,14 +296,18 @@ func (l *Limiter) BatchSpend(ctx context.Context, txns []Transaction) (*Decision
 		}
 
 		if d.allowed && (tat != d.newTAT) && txn.spend {
-			// New bucket state should be persisted.
-			if bucketExists {
+			if !bucketExists {
+				// No bucket exists, initialize a new bucket with BatchSetNotExist.
```

Comment:
This is a "how" comment, and it specifies implementation behavior where the implementation is several lines below.

I'd prefer a "why" comment down at line 328, e.g.

"Use `BatchSetNotExisting` so that we detect if concurrent requests have initialized this bucket at the same time, which would result in overwriting if we used a plain "SET" command. If that happens, fall back to incrementing."